File: src/pattern/visitor/java/Entry.java
package pattern.visitor.java;

public abstract class Entry implements Element{
    abstract String getName();
    abstract int getSize();
    public Entry add(Entry entry){
        throw new FileTreatmentException();
    }

    @Override
    public String toString() {
        return this.getName()+"("+this.getSize()+")";
    }
}

File: src/pattern/visitor/kotlin/Entry.kt
package pattern.visitor.kotlin

abstract class Entry: Element{
    abstract fun getName(): String
    abstract fun getSize(): Int
    open fun iterator(): Iterator<Element>{
        throw FileTreatmentException()
    }
    open fun add(entry: Entry){
        throw FileTreatmentException()
    }

    override fun toString(): String {
        return "${this.getName()}(${this.getSize()})"
    }
}